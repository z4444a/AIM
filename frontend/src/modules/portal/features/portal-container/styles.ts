import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';
import { WithStyles } from '@material-ui/core/styles/withStyles';
import { StyleRules } from '@material-ui/core/styles';

export const styles = (theme: Theme): StyleRules =>
  createStyles({
    content: {
      flexGrow: 1,
    },
    toolbar: theme.mixins.toolbar,
    mainContainer: {
      width: '100vw',
      height: '100vh',
    },
    bodyContainer: {
      display: 'flex',
      height: 'calc(100% - 64px)',
      marginTop: '64px',
    },
  });

export type Style = WithStyles<'content' | 'toolbar' | 'mainContainer' | 'bodyContainer'>;
