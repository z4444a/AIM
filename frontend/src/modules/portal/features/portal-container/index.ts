import withStyles from '@material-ui/core/styles/withStyles';
import { PortalContainer } from './portal-container';
import { styles } from './styles';

export default withStyles(styles)(PortalContainer);
