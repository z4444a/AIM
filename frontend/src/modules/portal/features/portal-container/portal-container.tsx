import React from 'react';
import Header from '../header/index';
import MuiBreadcrumbs from '../../../../commons/components/breadcrumbs/mui-breadcrumbs';
import AuthGuard from '../auth-guard/index';
import { Style } from './styles';
import { Redirect, Route, Switch } from 'react-router';
import { Path } from '../../../../commons/path';
import Resources from '../../../resources';
import Chart from '../../../resource-pool/chart/chart';
import RouteProtector from '../route-protector/route-protector';
import { Role } from '../../../../commons/role';
import EmployeesPage from '../../../employees/features/employees-page/index';
import SideBar from '../side-bar/index';
import ResourcePoolPage from '../../../resource-pool/features/resource-pool-grid-page/index';
import ResourceTypesPage from '../../../resource-types/features/resource-types-page/index';
import RequestsPage from '../../../requests/features/requests-page/index';
import TypeCreatePage from '../../../resource-types/features/create-type-page/index';
import UpdateTypePage from '../../../resource-types/features/update-type-page/index';
import RequestCreatePage from '../../../requests/features/request-create-page/index';
import ResourcePoolCreatePage from '../../../resource-pool/features/resource-pool-create-page/index';
import ResourcePoolUpdatePage from '../../../resource-pool/features/resource-pool-update-page/index';
import RequestAcceptPage from '../../../requests/features/request-accept-page';
import EmployeeChangeRolePage from '../../../employees/features/employee-change-role-page/index';
import SettingsPage from '../../../settings/feature/settings-page/index';

export type Props = Style;

export class PortalContainer extends React.Component<Props> {
  public render(): React.ReactNode {
    const { classes } = this.props;
    return (
      <AuthGuard>
        <div className={classes.mainContainer}>
          <Header>
            <MuiBreadcrumbs />
          </Header>
          <div className={classes.bodyContainer}>
            <SideBar />
            <main className={classes.content}>{this.getRoutes()}</main>
          </div>
        </div>
      </AuthGuard>
    );
  }

  private getRoutes(): React.ReactNode {
    return (
      <Switch>
        <Redirect exact path={Path.DEFAULT} to={Path.REQUESTS} />
        <Route exact path={Path.REQUEST_CREATE} component={RequestCreatePage} />
        <Route exact path={Path.REQUEST_CARD} component={RequestAcceptPage} />
        <Route exact path={Path.REQUESTS} component={RequestsPage} />
        <Route exact path={Path.RESOURCES} component={Resources} />
        <RouteProtector requiredRoles={[Role.ADMIN, Role.POOL_CREATOR, Role.POOL_OWNER]}>
          <Route exact path={Path.RESOURCE_POOL_CHART} component={Chart} />
          <Route exact path={Path.RESOURCE_POOLS} component={ResourcePoolPage} />
          <RouteProtector requiredRoles={[Role.ADMIN, Role.POOL_CREATOR]}>
            <Route exact path={Path.RESOURCE_POOL_CREATE} component={ResourcePoolCreatePage} />
            <Route exact path={Path.RESOURCE_POOL_UPDATE} component={ResourcePoolUpdatePage} />
            <RouteProtector requiredRoles={[Role.ADMIN]}>
              <Route exact path={Path.RESOURCE_TYPE_CREATE} component={TypeCreatePage} />
              <Route exact path={Path.RESOURCE_TYPE_UPDATE} component={UpdateTypePage} />
              <Route exact path={Path.RESOURCE_TYPES} component={ResourceTypesPage} />
              <Route exact path={Path.EMPLOYEE_CARD} component={EmployeeChangeRolePage} />
              <Route exact path={Path.EMPLOYEES} component={EmployeesPage} />
              <Route exact path={Path.SETTINGS} component={SettingsPage} />
              <Redirect path={Path.DEFAULT} to={Path.REQUESTS} />
            </RouteProtector>
            <Redirect path={Path.DEFAULT} to={Path.REQUESTS} />
          </RouteProtector>
          <Redirect path={Path.DEFAULT} to={Path.REQUESTS} />
        </RouteProtector>
        <Redirect path={Path.DEFAULT} to={Path.REQUESTS} />
      </Switch>
    );
  }
}
