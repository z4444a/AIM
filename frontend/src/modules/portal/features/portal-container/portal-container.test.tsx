import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { PortalContainer, Props } from './portal-container';

describe('PortalContainer', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const props: Props = {
      classes: {
        content: 'content',
        toolbar: 'toolbar',
        mainContainer: 'mainContainer',
        bodyContainer: 'bodyContainer',
      },
    };
    wrapper = shallow(<PortalContainer {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
