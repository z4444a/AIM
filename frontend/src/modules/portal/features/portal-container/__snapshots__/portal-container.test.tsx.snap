// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortalContainer should render correctly 1`] = `
<Connect(withRouter(AuthGuard))>
  <div
    className="mainContainer"
  >
    <Connect(WithStyles(Header))>
      <withRouter() />
    </Connect(WithStyles(Header))>
    <div
      className="bodyContainer"
    >
      <Connect(WithStyles(SideBar)) />
      <main
        className="content"
      >
        <Switch>
          <Redirect
            exact={true}
            path="/"
            push={false}
            to="/requests"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/requests/create"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/requests/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/requests"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/resources"
          />
          <Connect(RouteProtector)
            requiredRoles={
              Array [
                "ADMIN",
                "POOL_CREATOR",
                "POOL_OWNER",
              ]
            }
          >
            <Route
              component={[Function]}
              exact={true}
              path="/resource-pools/chart"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/resource-pools"
            />
            <Connect(RouteProtector)
              requiredRoles={
                Array [
                  "ADMIN",
                  "POOL_CREATOR",
                ]
              }
            >
              <Route
                component={[Function]}
                exact={true}
                path="/resource-pools/create"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/resource-pools/:id"
              />
              <Connect(RouteProtector)
                requiredRoles={
                  Array [
                    "ADMIN",
                  ]
                }
              >
                <Route
                  component={[Function]}
                  exact={true}
                  path="/resource-types/create"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/resource-types/:id"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/resource-types"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/employees/:id"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/employees"
                />
                <Route
                  component={[Function]}
                  exact={true}
                  path="/settings"
                />
                <Redirect
                  path="/"
                  push={false}
                  to="/requests"
                />
              </Connect(RouteProtector)>
              <Redirect
                path="/"
                push={false}
                to="/requests"
              />
            </Connect(RouteProtector)>
            <Redirect
              path="/"
              push={false}
              to="/requests"
            />
          </Connect(RouteProtector)>
          <Redirect
            path="/"
            push={false}
            to="/requests"
          />
        </Switch>
      </main>
    </div>
  </div>
</Connect(withRouter(AuthGuard))>
`;
